flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/SourceTaskTerminationTest.java

```diff
@@ -122,6 +124,26 @@ private void stopWithSavepointStreamTaskTestHelper(final boolean withMaxWatermar
 		executionThread.join();
 	}
 
+	private void waitForLatchSetByTask(OneShotLatch latch, StreamTaskTestHarness<?> srcTaskTestHarness) throws Exception {
+		final StreamTask<?, ?> srcTask = srcTaskTestHarness.getTask();
+		final Thread executionThread = srcTaskTestHarness.taskThread;
+		while (executionThread.isAlive() && srcTask.isRunning()) {
```

Comment:
I guess we can merge the following `while` and `if` logics into this `while` to avoid judging `srcTask.isRunning()` three times separately, and make the logics more close with each other.